### PR TITLE
feat: SIGINT-1419 GitHub integration

### PR DESCRIPTION
**Pull request by jahid1209**
- Added github cloud and enterprise server integration.
- Added Blackduck PR comment support.
- Added coverity PR comment support.
- Added/modified unit tests.
- Tested BD PR comment for both github cloud and enterprise server.
- TODO: Coverity PR comment for both cloud and enterprise server.

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/synopsys/security/scan/global/LogMessages.java [hunk 1]
```diff
@@ -12,7 +12,8 @@ public class LogMessages {
     public static final String COVERITY_PARAMETER_VALIDATION_FAILED = "Coverity parameters are not valid";
     public static final String POLARIS_PARAMETER_VALIDATION_FAILED = "Polaris parameters are not valid";
     public static final String NO_BITBUCKET_TOKEN_FOUND =
-            "PrComment or FixPr is set true but no bitbucket token found!";
+            "PrComment or FixPr is set true but no Bitbucket token found!";
+    public static final String NO_GITHUB_TOKEN_FOUND = "PrComment or FixPr is set true but no Github token found!";
```
[the remark this answers is not in the document]
Comment (jahid1209) [reply src/main/java/io/jenkins/plugins/synopsys/security/scan/global/LogMessages.java:16]: I don't know why we kept 'FixPR' in the first place!

File: src/main/java/io/jenkins/plugins/synopsys/security/scan/service/scm/github/GithubRepositoryService.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import io.jenkins.plugins.synopsys.security.scan.global.Utility;
+import io.jenkins.plugins.synopsys.security.scan.input.github.Github;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Map;
+
+public class GithubRepositoryService {
+    private final LoggerWrapper logger;
+    private String GITHUB_CLOUD_HOST_URL = "https://github.com/";
+    private String INVALID_GITHUB_REPO_URL = "Invalid Github repository URL";
```
[the remark this answers is not in the document]
Comment (jahid1209) [reply src/main/java/io/jenkins/plugins/synopsys/security/scan/service/scm/github/GithubRepositoryService.java:17]: Umm, if the fields are `final`, then `spotless` will complain about making them `static`. And I don't wanna make them `static`.

**Review by zaman-akib — APPROVED** [review]

**Comment by kishorikumar** [comment]
@jahid1209 Please make sure, we have tested the Github enterprise server configuration as well.

**Comment by jahid1209** [comment]
@kishorikumar Yes, we did test in the Github enterprise.

**Pull request merged**